src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,2233 @@
+      step_mem->cvals = NULL;
+      ark_mem->lrw -= step_mem->nfusedopvecs;
+    }
+    if (step_mem->Xvecs != NULL)
+    {
+      free(step_mem->Xvecs);
+      step_mem->Xvecs = NULL;
+      ark_mem->liw -= step_mem->nfusedopvecs;
+    }
+
+    /* free the time stepper module itself */
+    free(ark_mem->step_mem);
+    ark_mem->step_mem = NULL;
+  }
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_PrintMem:
+
+  This routine outputs the memory from the LSRKStep structure to
+  a specified file pointer (useful when debugging).
+  ---------------------------------------------------------------*/
+void lsrkStep_PrintMem(ARKodeMem ark_mem, FILE* outfile)
+{
+  ARKodeLSRKStepMem step_mem;
+  int retval;
+
+  /* access ARKodeLSRKStepMem structure */
+  retval = lsrkStep_AccessStepMem(ark_mem, __func__, &step_mem);
+  if (retval != ARK_SUCCESS) { return; }
+
+  /* print integrator memory to file */
+  switch (step_mem->LSRKmethod)
+  {
+  case ARKODE_LSRK_RKC_2:
+    fprintf(outfile, "LSRKStep RKC time step module memory:\n");
+    break;
+  case ARKODE_LSRK_RKL_2:
+    fprintf(outfile, "LSRKStep RKL time step module memory:\n");
+    break;
+  case ARKODE_LSRK_SSP_S_2:
+    fprintf(outfile, "LSRKStep SSP(s,2) time step module memory:\n");
+    break;
+  case ARKODE_LSRK_SSP_S_3:
+    fprintf(outfile, "LSRKStep SSP(s,3) time step module memory:\n");
+    break;
+  case ARKODE_LSRK_SSP_10_4:
+    fprintf(outfile, "LSRKStep SSP(10,4) time step module memory:\n");
+    break;
+  default:
+    arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                    "Invalid method option.");
+    return;
+  }
+
+  fprintf(outfile, "LSRKStep: q                   = %i\n", step_mem->q);
+  fprintf(outfile, "LSRKStep: p                   = %i\n", step_mem->p);
+
+  if (step_mem->is_SSP)
+  {
+    fprintf(outfile, "LSRKStep: req_stages          = %i\n",
+            step_mem->req_stages);
+    fprintf(outfile, "LSRKStep: nfe                 = %li\n", step_mem->nfe);
+  }
+  else if (!step_mem->is_SSP)
+  {
+    /* output integer quantities */
+    fprintf(outfile, "LSRKStep: req_stages            = %i\n",
+            step_mem->req_stages);
+    fprintf(outfile, "LSRKStep: dom_eig_nst           = %li\n",
+            step_mem->dom_eig_nst);
+    fprintf(outfile, "LSRKStep: stage_max             = %i\n",
+            step_mem->stage_max);
+    fprintf(outfile, "LSRKStep: stage_max_limit       = %i\n",
+            step_mem->stage_max_limit);
+    fprintf(outfile, "LSRKStep: dom_eig_freq          = %i\n",
+            step_mem->dom_eig_freq);
+
+    /* output long integer quantities */
+    fprintf(outfile, "LSRKStep: nfe                   = %li\n", step_mem->nfe);
+    fprintf(outfile, "LSRKStep: dom_eig_num_evals     = %li\n",
+            step_mem->dom_eig_num_evals);
+
+    /* output sunrealtype quantities */
+    fprintf(outfile, "LSRKStep: dom_eig               = %" RSYM " + i%" RSYM "\n",
```

Comment:
Using `+` in the format specifier will ensure the sign is always for the imaginary part
```suggestion
    fprintf(outfile, "LSRKStep: dom_eig               = %" RSYM " %+" RSYM "i\n",
```